### PR TITLE
fix: use isPropValid when considering forwarding props

### PR DESCRIPTION
**Pull request by mzikherman**
This might be more futureproof.

### CODE REVIEW

File: src/System/Components/RouterLink.tsx [hunk 1]
```diff
@@ -121,7 +122,6 @@ const VALID_ROUTER_LINK_PROPS = [
   "data-test",
   "data-testid",
   "exact",
-  "onClick",
```
Comment (mzikherman) [inline src/System/Components/RouterLink.tsx:124]: Looks like most of these can be removed as well when using `isPropValid` - https://github.com/emotion-js/emotion/blob/main/packages/is-prop-valid/src/index.ts includes `children`, `style`, `target`, etc.

Comment (anandaroop) [reply src/System/Components/RouterLink.tsx:124]: Looks like everything except `activeClassName`, `activeStyle`, `exact` — given that it blanket includes [`on*`](https://github.com/emotion-js/emotion/blob/e310c6e6a9f0b4f55030831ef65cd07c7105ae01/packages/is-prop-valid/src/index.ts#L12-L13) and [`data*` and `aria*`](https://github.com/emotion-js/emotion/blob/e310c6e6a9f0b4f55030831ef65cd07c7105ae01/packages/is-prop-valid/src/props.js#L499)

**Review by anandaroop — DISMISSED** [review]
I like it — should we [chop that list](https://github.com/artsy/force/pull/14691#discussion_r1811251908) even further and then let this go through another full Integrity run?

File: src/System/Components/RouterLink.tsx [hunk 2]
```diff
@@ -114,18 +115,7 @@ type RouterLinkMixinProps = BoxProps & {
...
-  "children",
-  "data-test",
-  "data-testid",
-  "exact",
-  "onClick",
-  "style",
-  "target",
-  "to",
-]
+const VALID_ROUTER_LINK_PROPS = ["activeClassName", "activeStyle", "exact"]
```
Comment (mzikherman) [inline src/System/Components/RouterLink.tsx:118]: Nice!

**Comment by anandaroop** [comment]
Integrity [likes it](https://app.circleci.com/pipelines/github/artsy/integrity/25748/workflows/da331ae0-070f-4741-a91e-6be0410067d3), so I will go ahead and deploy this change

**Pull request merged**